### PR TITLE
Adding a tip on the merge case with a variable

### PR DESCRIPTION
**Pull request by mikeSimonson**

### CODE REVIEW

File: doc/filters/merge.rst [hunk 1]
```diff
@@ -40,6 +40,20 @@ overridden.
...
+.. tip::
+
+    If the values or keys you want to merge are variable, you need to enclose them into parenthesis :
+
+    .. code-block:: jinja
+        {% set mergedItems = {} %}
+        {% set items = { 'apple': 'fruit', 'orange': 'fruit' } %}
+
+        {% for key, value in items %} 
+            {% set mergedItems = mergedItems|merge({ (key): (value) }) %}
```
Comment (fabpot) [inline doc/filters/merge.rst]: Parenthesis are only needed for keys and as this is already documented when we talk about arrays, I don't see the need to repeat it here.

Comment (mikeSimonson) [inline doc/filters/merge.rst]: Ok. Care on a note about the array_merge particular behavior with numeric variable as key ?

Comment (fabpot) [inline doc/filters/merge.rst]: It's already implicit in the note below about the fact that `array_merge` is used behind the scene. But we can indeed add a sentence in the note about how that works. Can you do that?

Comment (mikeSimonson) [inline doc/filters/merge.rst]: done in #1715

**Pull request closed without merging**